LucasTerra.lua

```diff
@@ -2512,37 +2512,54 @@ end
 -- @param	type	Type can be any alarm name you found on the alarms engine.
 -- @param	playsound	Set 'no' or 'yes' to play sound.
 -- @param	pausebot	Set 'no' or 'yes' to pause bot.
--- @param	logout	Set 'no' or 'yes' to logout.
+-- @param	logout		Set 'no' / 'Logout' / 'Force Logout' / 'Close Client' to logout.
+-- @param	cloudmsg	Set 'no' or 'yes' to send cloud message when alert is triggered.
 -- @returns void
 
-function setalarm(alarmtype, playsound, pausebot, logout, t)
-	local alarmtypes = {"PlayerOnScreen", "PlayerAttacking", "MonsterAttacking", "PrivateMessage", "DefaultMessage", "GMDetected", "Disconnected", "CharacterStuck", "HealthBelow", "ManaBelow", "UnjustKill", "EnemiesOnline"}
-	local alarmtypesLower = alarmtypes
-	table.tolower(alarmtypesLower)
-	local pos = table.find(alarmtypesLower,alarmtype:lower())
-	if not pos then
+function setalarm(alarmType, playSound, pauseBot, logout, cloudMsg)
+	local alarmTypes = {"PlayerOnScreen", "PlayerAttacking", "MonsterAttacking", "PrivateMessage", "DefaultMessage", "GMDetected", "Disconnected", "CharacterStuck", "HealthBelow", "ManaBelow", "UnjustKill", "EnemiesOnline"}
+	alarmType = alarmType:lower()
+
+	local foundAlarmType = nil
+	for _, v in ipairs(alarmTypes) do
+		if alarmType == v:lower() then
+			foundAlarmType = v
+			break
+		end
+	end
+	
+	if not foundAlarmType then
 		return
 	end
 
-	if not playsound or playsound == 'off' or playsound == 'no' or playsound == 0 then
-		playsound = 'no'
-	elseif playsound == 'on' or playsound == 'yes' or playsound == 1 then
-		playsound = 'yes'
-	end
-	if not pausebot or pausebot == 'off' or pausebot == 'no' or pausebot == 0 then
-		pausebot = 'no'
-	elseif pausebot == 'on' or pausebot == 'yes' or pausebot == 1 then
-		pausebot = 'yes'
+	local logoutAction
+	if type(logout) == 'string' then
+		logout = logout:lower()
+
+		if logout:find('force') then
+			logoutAction = 'Force Logout'
+		elseif logout:find('close') then
+			logoutAction = 'Close Client'
+		elseif logout == 'logout' or logout == 'yes' then
+			logoutAction = 'Logout'
+		end
+	elseif type(logout) == 'number' then

```

Comment:
I'm not sure if this is a good idea. Specially since 1 (which would usually be considered as a truthful value) is translated into 'no'.

@XtrmJosh @leonardofalk @lucasterra Care to give some input?